### ISSUE TITLE
FR: Copy title to aliases

### ISSUE THREAD

**Issue by chmac** [platers/obsidian-linter#1005]
## Is Your Feature Request Related to a Problem? Please Describe.

I use the `title` YAML field to publish notes. But in Obsidian, only `aliases` are searchable. So I need to manually copy `title` into `aliases` (as I don't have a leading `h1` which has my title).

## Describe the Solution You'd Like

It would be great to have an option for the "YAML Title Alias" section which sources the title from the `title` frontmatter key.

**Please include an example where applicable:**

Example filename: `foo.md`

``` markdown
---
title: Bar
---
```

``` markdown
---
title: Bar
aliases:
  - Bar
---
```

## Describe Alternatives You've Considered

I'm not aware of any alternative strategy to achieve this, but I'd be happy to hear suggestions if there is a viable workaround. :-)

## Additional Context

Thanks for Linter. Took me a while to figure out what this plugin could do as I assumed it was mostly linting, but it's incredibly powerful, and super useful coming from dendron into obsidian. 🧡

**Comment by hkcomori** [comment]
I want this feature very much too.

If we only write `H1`, it will work fine.
Then, "YAML Title" creates `title` frontmatter key and "YAML Title Alias" adds same value to `aliases` frontmatter key.

But this alternative cannot be used if H1 gets in the way in the post-process.

**Comment by pjkaufman** [comment]
Hey @chmac , @hkcomori , YAML Title currently has a mode that determines how it functions (see [https://platers.github.io/obsidian-linter/settings/yaml-rules/#yaml-title](https://platers.github.io/obsidian-linter/settings/yaml-rules/#yaml-title)). Would it make sense to just add a mode to `YAML Title Alias` to allow for this functionality where you can specify that you want the title alias to come from either the H1, the filename, or how it currently works? That seems the simplest change to me which would make it easier to maintain (hopefully) and get this added faster.

What do y'all think?

**Comment by hkcomori** [comment]
@pjkaufman
Yes, it would be great if the option could be one of `Title key`, `First H1`, `Filename`, `Title key or First H1 if Title key is missing or Filename if H1 is missing` for `YAML Title Alias`.
The `Title Key` means the value specified in `YAML Title`.
I think variable would match more use cases than the fixed key name `title`.

**Comment by chmac** [comment]
@pjkaufman I'm running into this again. Would you be open to taking a PR? I had a quick look and it seems like it should be fairly easy to add this into here:

https://github.com/platers/obsidian-linter/blob/master/src/rules/yaml-title-alias.ts

**Comment by chmac** [comment]
I’ve been looking through the code a bit, and my first thought would be to add a check for an existing title property around here:
https://github.com/platers/obsidian-linter/blob/1877e2bbc4499734ede092a4b4e006d2968e4648/src/rules/yaml-title-alias.ts#L129-L136

If a title already exists in the frontmatter, we could escape it and return both escaped and unescaped versions early.

Conceptually, it makes sense that if a frontmatter title is already defined, it should take precedence over the first H1 or the filename — it’s a more explicit statement of what the note’s title should be.

However, this would change existing behavior and could break workflows that currently depend on the H1/filename priority. To avoid that, we could introduce an option such as “Ignore frontmatter title” or “Don’t use frontmatter title.” The default value would determine whether existing setups continue working as before. If we want to preserve current behavior, the default should be true; if we want to treat frontmatter titles as preferred, it should be false. That choice would need to be decided.

At present, users don’t have any control over where the title is sourced — it’s always H1 or filename. Allowing frontmatter title support would make this more flexible. There are a few edge cases (for example, users who define a frontmatter title but still want the H1 added to aliases), but an option flag should be enough to handle those without unnecessary complexity.

A more advanced approach could be a dropdown letting users choose the title source, but that’s probably beyond what I’d plan to include in an initial PR.
